### ISSUE TITLE
sortable grid layout issue if mathjax used

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#933]
I had it reported to me that there's an issue with the layout of the sortable grid page. It turns out this possibly only occurs when mathjax is used in the moveable tiles rather than plain text. Here's an example:

Page 1 with mathjax https://xot.xerte.org.uk/play.php?template_id=218#resume=1 
Page 2 with text rather than mathjax https://xot.xerte.org.uk/play.php?template_id=218#resume=2
Page 3 with mathjax and text in case the text items made a difference https://xot.xerte.org.uk/play.php?template_id=218#resume=3

**Comment by JohnSmith-LT** [comment]
Hi @ronm123 I patched this back in July but it's obviously not made it onto that server yet

https://github.com/thexerteproject/xerteonlinetoolkits/commit/15f6d7bdcebb87b80c7cd464e8a9e225281efb50

**Comment by JohnSmith-LT** [comment]
I think this does beag the question: why do we have to jusmp through so many hoops with MathJax and why does it have to be continually re-rendered manually?

For example, on JQuery-UI page they have a draggable. I edited it to include \[Q=v/w\] and then loaded mathjax script (smae one we use in Xerte) via $.getScript().. The result is this:

![MathJaxNoRerender](https://user-images.githubusercontent.com/3907362/91234834-12e81b80-e72c-11ea-9ad2-6a7a918919c3.gif)

So the drag is absolutely fine, however in Xerte, as soon as we drag we get this:

![XerteMathJaxRerender](https://user-images.githubusercontent.com/3907362/91234946-5478c680-e72c-11ea-856e-99bff0226f5f.gif)

So what are we doing differently? Any ideas @FayCross  or @torinfo or is it simply down to the version that we are using? If we could fix that then it would simplify a lot of page models that are doing this continually...

**Comment by ronm123** [comment]
@JohnSmith-LT re your previous comment about having patched this: I know you patched the issue with the mathjax showing code instead of mathjax and you're right that probably just isn't part of xot.xerte.org.uk yet but the issue I raised here is different and perhaps I didn't make that clear enough :-( This was about the layout not the rendering of the mathjax. In the screenshot below (from the example linked above) the first page has the same left hand Parameter text as the second page but the first page has mathjax in the units and data columns whereas the second page has plain text.  For some reason this not only affects the width of the units and data columns but also the parameter column too. e.g. the first is all scrunched up to the right and the second is spaced how you might expect. I guess it's to do with how the widths are dynamically calculated but when viewed on a reasonable sized screen and especially a large screen the layout when mathjax is used seems wrong?

**Comment by ronm123** [comment]
BTW obviously there are obviously two slightly separate topics here now e.g.
1. layout of the sortable grid page especially when mathjax is used.
2. issues with mathjax more generally and why it doesn't work in xerte like it does in the jquery example.

**Comment by JohnSmith-LT** [comment]
Ah ok, as soon as you mentioned MathJax I focussed on that, so this is what you mean (with the LO open full screen or on a larger monitor it isn't so obvious, yes the grid is bigger but at least all the text is viewable)

![image](https://user-images.githubusercontent.com/3907362/91283565-a8ae9580-e782-11ea-94df-3d76f64823a8.png)

I don't think it's the MathJax per se, I think it's more to do with the calculations to set the column widths that is off and perhaps the resize handler. For example, if I view page 2 and then turn fullscreen off I get the above but if I view page 1 then turn fullscreen off and then go to page 2 I get this, which seems fine...

![image](https://user-images.githubusercontent.com/3907362/91285183-ca108100-e784-11ea-8f79-7f4e76f3f960.png)

**Comment by ronm123** [comment]
Sorry @JohnSmith-LT I forgot to post my screenshot in my previous post :-(
![image](https://user-images.githubusercontent.com/5167146/91292920-26789e00-e78f-11ea-8d9a-76487c51b965.png)
That's what I meant about the difference between whether mathjax is included or not.

**Comment by JohnSmith-LT** [comment]
Yes, I realised when I was moving the other stuff that what you could be referring to was the 1st column, the labels which is different in both cases. Suspect it's something to do with the size of the other columns though and the calculations - would be interesting I think to see a box with both MathJax and a really long word in a single box to see what happens in that case, if that's easy for you to do?

**Comment by ronm123** [comment]
@JohnSmith-LT @FayCross I'm wondering if the mathjax issue I reported previously here is separate from the alignment issue. I had an example shared with me that doesn't use mathjax and only has two columns but the layout is clearly wrong when the Text Alignment option is set to Left rather than Right or Top. I can't spot where to fix this so I've put a copy on xot.xerte and you both have co-author access. However to clarify this seems to only be an issue when the text align option is set to left e.g.
![image](https://user-images.githubusercontent.com/5167146/94257059-a460ce00-ff22-11ea-8430-27bee10222f7.png)

You can see the impact via https://xot.xerte.org.uk/play.php?template_id=225#page1 where page 1 is set to Right and layout is fine. Page 2 is set to Top and again layout is fine. Page 3 is set to left (which is the default) and the table is all scrunched over to the right and about a 3rd of the width or less of the other alignment modes,
